source/mainnet/net/references/grpc.rst

```diff
@@ -0,0 +1,886 @@
+
+   .. collapse:: View JSON schema
+
+      .. literalinclude:: grpc-json-schemas/GetAccountNonFinalized.json
+         :language: json
+
+.. function:: GetNextAccountNonce(accountAddress) -> ?AccountNonceResponse
+
+   Returns the next available nonce for this account.
+
+   :param accountAddress: The account to query.
+   :type accountAddress: |grpc-account-address|_
+   :returns: An account nonce and whether there are any non-finalized
+             transactions for the account.
+   :rtype: ``?AccountNonceResponse`` (see JSON schema below)
+
+   .. collapse:: View JSON schema
+
+      .. literalinclude:: grpc-json-schemas/GetNextAccountNonce.json
+         :language: json
+
+Smart contracts
+===============
+
+.. function:: GetModuleList(blockHash) -> ?[ModuleHash]
+
+   Get a list of all smart contract modules that existed when the given block was created.
+
+   :param blockHash: The given block.
+   :type blockHash: |grpc-block-hash|_
+   :returns: A list of hashes of smart contract modules.
+   :rtype: ``[ModuleHash]`` (see JSON schema below)
+
+   .. collapse:: View JSON schema
+
+      .. literalinclude:: grpc-json-schemas/GetModuleList.json
+         :language: json
+
+.. function:: GetModuleSource(blockHash, moduleReference) -> ?[Byte]
+
+   Get the binary source of a smart contract module.
+
+   :param blockHash: The given block.
+   :type blockHash: |grpc-block-hash|_
+   :param moduleReference: The reference (hash) of the smart contract module.
+   :type blockHash: |grpc-module-reference|_
+   :returns: The binary source of the module.
+   :rtype: ``[Byte]``
+
+.. _grpc-get-instances:
+
+.. function:: GetInstances(blockHash) -> ?[ContractAddress]
+
+   Get a list of all smart contract instances that existed when the given block
+   was created.
+
+   :param blockHash: The given block.
+   :type blockHash: |grpc-block-hash|_
+   :returns: A list of smart contract addresses.
+   :rtype: ``[ContractAddress]`` (see JSON schema below)
+
+   .. collapse:: View JSON schema
+
+      .. literalinclude:: grpc-json-schemas/GetInstances.json
+         :language: json
+
+.. _grpc-get-instance-info:
+
+.. function:: GetInstanceInfo(blockHash, contractAddress) -> ?InstanceInfo
+
+   Get information about the given smart contract instance in the given block.
+
+   :param blockHash: The given block.
+   :type blockHash: |grpc-block-hash|_
+   :param contractAddress: The smart contract instance.
+   :type contractAddress: |grpc-contract-address|_
+   :returns: Information about the smart contract instance.
+   :rtype: ``?InstanceInfo`` (see JSON schema below)
+
+   .. collapse:: View JSON schema
+
+      .. literalinclude:: grpc-json-schemas/GetInstanceInfo.json
+         :language: json
+
+.. function:: InvokeContract(blockHash, contractContext) -> ?InvokeContractResult
+
+   Invoke a smart contract instance and view its results as if it had been
+   updated at the end of the given block. *This is not a transaction*.
```

Comment:
I would recommend elaborating on "This is not a transaction".

Reply:
I've added some more text. Let me know if that suffices.